fuse/utils/ndict.py

```diff
@@ -111,23 +111,41 @@ def flatten(self) -> dict:
         #you can use it to get a list of the flat keys:
         print(nx.flatten().keys())
         """
+        flat_dict = {}
+        NDict._flatten_static(self._stored, None, flat_dict)
+        return flat_dict
 
-        all_keys = {}
-        for key in self._stored:
-            if isinstance(self._stored[key], MutableMapping):
-                all_sub_keys = NDict(self[key]).flatten()
-                keys_to_add = {f"{key}.{sub_key}": all_sub_keys[sub_key] for sub_key in all_sub_keys}
-                all_keys.update(keys_to_add)
-            else:
-                all_keys[key] = self._stored[key]
-
-        return all_keys
+    @staticmethod
+    def _flatten_static(item: Union[dict, Any], prefix: str, flat_dict: dict) -> None:
+        if isinstance(item, MutableMapping):
+            for key, value in item.items():
+                if prefix is None:
+                    cur_prefix = key
+                else:
+                    cur_prefix = f"{prefix}.{key}"
+                NDict._flatten_static(value, cur_prefix, flat_dict)
+        else:
+            flat_dict[prefix] = item
 
     def keypaths(self) -> List[str]:
         """
         returns a list of keypaths (i.e. "a.b.c.d") to all values in the nested dict
         """
-        return list(self.flatten().keys())
+        return NDict._keypaths_static(self._stored, None)
```

Comment:
Why not using the same paradigm as before? Just calling `flatten()`? 
The two static functions has a lot in common

Reply:
Mostly cause I don't want the overhead of creating a dictionary and extracting the keys,